--- a/src/dhcp-manager/nm-dhcp-dhclient.c	2010-05-22 22:35:03.000000000 +0400
+++ b/src/dhcp-manager/nm-dhcp-dhclient.c	2010-06-25 00:22:47.618906714 +0400
@@ -64,7 +64,8 @@
 const char *
 nm_dhcp_dhclient_get_path (const char *try_first)
 {
-	static const char *dhclient_paths[] = {
+	return NULL; // Never use dhclient
+/*	static const char *dhclient_paths[] = {
 		"/sbin/dhclient",
 		"/usr/sbin/dhclient",
 		"/usr/pkg/sbin/dhclient",
@@ -83,6 +84,7 @@
 	}
 
 	return *path;
+	*/
 }
 
 static char *
