--- bumblebee-2.3.1.orig/stages/setvars	2011-09-03 01:42:46.257704077 +0400
+++ bumblebee-2.3.1/stages/setvars	2011-09-03 01:57:15.619667543 +0400
@@ -21,12 +21,14 @@
 DESTDIR=
 
 # Install locations
-BINDIR=/usr/local/bin
-SBINDIR=/usr/local/sbin
+BINDIR=/usr/bin
+SBINDIR=/usr/sbin
 CONFDIR=/etc/bumblebee
-LIBDIR=/usr/local/lib/bumblebee
+LIBDIR=/usr/lib/bumblebee
+[[ $ARCH == x86_64 ]] && LIBDIR=/usr/lib64/bumblebee
 INITDIR=/etc/init.d
 XORG_MODULEPATH=/usr/lib/xorg/modules
+[[ $ARCH == x86_64 ]] && XORG_MODULEPATH=/usr/lib64/xorg/modules
 
 # available drivers, add nouveau in the future
 AVAILABLE_DRIVERS=('nvidia')
